include/zephyr/bluetooth/audio/bap.h

```diff
@@ -83,6 +83,85 @@ enum bt_bap_ep_state {
 	BT_BAP_EP_STATE_RELEASING = 0x06,
 };
 
+/**
+ * @brief Response Status Code
+ *
+ * These are sent by the server to the client when a stream operation is
+ * requested.
+ */
+enum bt_bap_ascs_rsp_code {
+	/** Server completed operation successfully */
+	BT_BAP_ASCS_RSP_CODE_SUCCESS = 0x00,
+	/** Server did not support operation by client */
+	BT_BAP_ASCS_RSP_CODE_NOT_SUPPORTED = 0x01,
+	/** Server rejected due to invalid operation length */
+	BT_BAP_ASCS_RSP_CODE_INVALID_LENGTH = 0x02,
+	/** Invalid ASE ID */
+	BT_BAP_ASCS_RSP_CODE_INVALID_ASE = 0x03,
+	/** Invalid ASE state */
+	BT_BAP_ASCS_RSP_CODE_INVALID_ASE_STATE = 0x04,
+	/** Invalid operation for direction */
+	BT_BAP_ASCS_RSP_CODE_INVALID_DIR = 0x05,
+	/** Capabilities not supported by server */
+	BT_BAP_ASCS_RSP_CODE_CAP_UNSUPPORTED = 0x06,
+	/** Configuration parameters not supported by server */
+	BT_BAP_ASCS_RSP_CODE_CONF_UNSUPPORTED = 0x07,
+	/** Configuration parameters rejected by server */
+	BT_BAP_ASCS_RSP_CODE_CONF_REJECTED = 0x08,
+	/** Invalid Configuration parameters */
+	BT_BAP_ASCS_RSP_CODE_CONF_INVALID = 0x09,
+	/** Unsupported metadata */
+	BT_BAP_ASCS_RSP_CODE_METADATA_UNSUPPORTED = 0x0a,
+	/** Metadata rejected by server */
+	BT_BAP_ASCS_RSP_CODE_METADATA_REJECTED = 0x0b,
+	/** Invalid metadata */
+	BT_BAP_ASCS_RSP_CODE_METADATA_INVALID = 0x0c,
+	/** Server has insufficient resources */
+	BT_BAP_ASCS_RSP_CODE_NO_MEM = 0x0d,
+	/** Unspecified error */
+	BT_BAP_ASCS_RSP_CODE_UNSPECIFIED = 0x0e,
+};
+
+/**
+ * @brief Response Reasons
+ *
+ * These are used if the @ref bt_bap_ascs_rsp_code value is
+ * @ref BT_BAP_ASCS_RSP_CODE_CONF_UNSUPPORTED, @ref BT_BAP_ASCS_RSP_CODE_CONF_REJECTED or
+ * @ref BT_BAP_ASCS_RSP_CODE_CONF_INVALID.
+ */
+enum bt_bap_ascs_reason {
+	/** No reason */
+	BT_BAP_ASCS_REASON_NONE = 0x00,
+	/** Codec ID */
+	BT_BAP_ASCS_REASON_CODEC = 0x01,
+	/** Codec configuration */
+	BT_BAP_ASCS_REASON_CODEC_DATA = 0x02,
+	/** SDU interval */
+	BT_BAP_ASCS_REASON_INTERVAL = 0x03,
+	/** Framing */
+	BT_BAP_ASCS_REASON_FRAMING = 0x04,
+	/** PHY */
+	BT_BAP_ASCS_REASON_PHY = 0x05,
+	/** Maximum SDU size*/
+	BT_BAP_ASCS_REASON_SDU = 0x06,
+	/** RTN */
+	BT_BAP_ASCS_REASON_RTN = 0x07,
+	/** Max transport latency */
+	BT_BAP_ASCS_REASON_LATENCY = 0x08,
+	/** Presendation delay */
+	BT_BAP_ASCS_REASON_PD = 0x09,
+	/** Invalid CIS mapping */
+	BT_BAP_ASCS_REASON_CIS = 0x0a,
+};
+
+/** ASE Control Point operation response  */
+struct bt_bap_ascs_rsp {
+	enum bt_bap_ascs_rsp_code code;
+	uint8_t reason;
```

Comment:
```suggestion
	enum bt_bap_ascs_reason reason;
```

Reply:
For the `reason` field we cannot use `enum bt_bap_ascs_reason`, because in case of an error during the Update Metadata operation we have to be able to set the reason field to value from `enum bt_audio_metadata_type` (test case ASCS/SR/SPE/BI-09-C).

Reply:
> For the `reason` field we cannot use `enum bt_bap_ascs_reason`, because in case of an error during the Update Metadata operation we have to be able to set the reason field to value from `enum bt_audio_metadata_type` (test case ASCS/SR/SPE/BI-09-C).

Union would work then

```suggestion
        union {
                enum bt_bap_ascs_reason reason;
                enum bt_audio_metadata_type metadata_type;
        };
```